_posts/pl/2021-12-XX-transactions.md

```diff
@@ -0,0 +1,217 @@
+---
+layout:    post
+title:     "Transakcje - co warto wiedzieć?"
+date:      2021-12-XX 08:00:00 +0100
+published: false
+didyouknow: false
+lang: pl
+author:    rmastalerek
+image:     /assets/img/posts/2021-12-14-transactions/transactions.jpg
+tags:
+    - transactions
+    - spring
+    - java
+    - transactional
+---
+
+Czy ktokolwiek, po zamówieniu w restauracji kawy i ciastka chciałby zapłacić za oba, a otrzymać tylko czarny, gorący napój? Czy do zaakceptowania byłby fakt, że po wysłaniu przelewu na pokaźną kwotę, uszczuplony zostałby jedynie nasz rachunek, a kwota nie zasiliłaby konta odbiorcy? Podane przykłady mają na celu lepsze zilustrowanie zagadnienia, które zostanie poruszone w niniejszym wpisie. Pewne rzeczy mają sens tylko wtedy, gdy wykonywane są kompleksowo. Tym razem chciałbym przedstawić krótkie wprowadzanie do transakcji. Wpis zostanie podzielony na 2 części. W części pierwszej skupię się na przybliżeniu samej definicji oraz zasady działania transakcyjności w systemach baz danych oraz wykorzystaniu adnotacji _@Transactional_ w Springu. Kolejna część poświęcona będzie problemom jakie napotkaliśmy w codziennej pracy, co pozwoliło je zidentyfikować oraz jak zostały wyeliminowane.
+
+# Transakcja
+Pewien zbiór operacji, które muszą zostać wykonane kompleksowo (wszystkie lub żadna), nazywamy wlaśnie **transakcją**. Odniesienie do operacji bankowej o takiej samej nazwie jest nieprzypadkowe. Wspomniany we wstępie przykład przelewu jest modelowym wzorem transakcji. Wykonując przelew w banku pieniądze muszą jedocześnie zostać odjęte z rachunku nadawcy, jak i dodane w tej samej kwocie do salda konta odbiorcy. Tylko wtedy transakcja kończy się sukcesem.
+
+W przypadku niepowodzenia, któregokolwiek z kroków, wykonane dotąd operacje muszą zostać cofnięte, a stan systemu przywrócony do chwili sprzed rozpoczęcia całej procedury.
+
+Transakcje cechują się zasadą ACID. Jest to akronim 5 właściwości (konkretnie ich angielskich nazw), którymi charaktryzuje się transakcja:
```

Comment:
ACID to raczej zbiór zasad a nie zasada i jest akronimem czterech a nie pięciu właściwości

Reply:
poprawione